x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/CostEntry.java

```diff
@@ -14,20 +14,32 @@
 import java.util.Map;
 
 final class CostEntry implements ToXContentObject {
-    final double costFactor;
+    final double usd_per_hour;
 
-    CostEntry(double costFactor) {
-        this.costFactor = costFactor;
+    CostEntry(double usdPerHour) {
+        this.usd_per_hour = usdPerHour;
     }
 
     public static CostEntry fromSource(Map<String, Object> source) {
-        return new CostEntry((Double) source.get("cost_factor"));
+        var val = source.get("usd_per_hour");
+
+        if (val instanceof Double d) {
+            return new CostEntry(d);
+        }
+
+        // Some JSON values have no decimal places and are passed in as Integers.
+        if (val instanceof Integer i) {
+            return new CostEntry(i.doubleValue());
+        }
+
+        // Likely an unexpected null value.
+        return new CostEntry(CostCalculator.DEFAULT_COST_USD_PER_CORE_HOUR * HostMetadata.DEFAULT_PROFILING_NUM_CORES);
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Semi-related: I looked around in the code but I believe that instances of this class are never rendered as XContent (i.e. in an API response) so we can just eliminate this method. When we do this, there is actually not much code left: we extract a double value from a map (with a default value for missing entries) so I wonder whether we can delete this class completely?

Reply:
:+1: removed